test/ng/locationSpec.js

```diff
@@ -2357,19 +2394,32 @@ describe('$location', function() {
       });
 
 
-      it('should only overwrite existing properties if values are boolean', function() {
+      it('should only overwrite existing properties if values are the correct type', function() {
         module(function($locationProvider) {
           $locationProvider.html5Mode({
             enabled: 'duh',
             requireBase: 'probably',
-            rewriteLinks: 'nope'
+            rewriteLinks: false
```

Comment:
This change doesn't test what it should. You need to supply an unsupported type and verify that it didn't get copied to the actual config.